Capabilities Cache

anthonydresser opened the pull request:
Cleans up how we handle cacheing capabilities for services and isolates it all to the capabilities service. Removes a lot of unnecessary code, while also fixing the bug that the connection wouldn't be buffered properly.

kburtram commented on src/sql/parts/connection/common/connectionConfig.ts:
```diff
@@ -29,32 +29,14 @@ export interface ISaveGroupResult {
-		}
-	}
-
-	public setCachedMetadata(cachedMetadata: sqlops.DataProtocolServerCapabilities[]): void {
```
I think @saurabh500 added these caches to help speed up startup.  Do you know if removing these will regress startup experience?

saurabh500 commented:
Cache was added because initial connection dialog was waiting for the services to come up to query capabilities. 
If this is removed then ops studio will startup and then no connection dialog box will popup until all the services are up.

kburtram commented on src/sql/parts/connection/common/connectionProfile.ts:
```diff
@@ -170,8 +170,19 @@ export class ConnectionProfile extends ProviderConnectionInfo implements interfa
 
-	public static createFromStoredProfile(profile: interfaces.IConnectionProfileStore, serverCapabilities: sqlops.DataProtocolServerCapabilities): ConnectionProfile {
-		let connectionInfo = new ConnectionProfile(serverCapabilities, undefined);
+	public static fromIConnectionProfile(capabilitiesService: ICapabilitiesService, profile: interfaces.IConnectionProfile) {
```
should this method have a return type?

kburtram commented:
Thanks @saurabh500! I setup a meeting on Monday to discuss if we want Connection Dialog to show automatically on 1st launch (among other 1st launch issues).  Assuming we do, then I think we'll need to fix that experience.

anthonydresser commented:
@saurabh500 as far as I understand, that functionality was in the protocol layer. It would pull the capabilities from the config.json instead of sending the request to the client. I removed that awhile ago as part of the 3.0 upgrade with the promise I would add it back in. I had a branch where I did that but I was waiting on a lot of my other changes to go in. This change is part of that larger change.

The caches I removed were simply caching the data that the capabilitiesService was getting, which it was getting from sending the request to the extension. So the caches I removed were basically just isolated caches. This changes just takes all the different places we were cacheing capabilities and isolates it all to the capabilitiesService itself.

AbbiePetcht commented on build/lib/i18n.js:
```diff
@@ -533,12 +531,9 @@ function getResource(sourceFile) {
         return { name: 'vs/workbench', project: workbenchProject };
     }
-
-    // {{SQL CARBON EDIT}}
```
>// {{SQL CARBON EDIT}} [](start = 4, length = 22)

why did you remove this line?

anthonydresser replied:
this is a compiled file. It compiled and got rid of it

saurabh500 commented:
I synced up with @anthonydresser offline and his improvements are different from the work that I had done. 
He will be working on the initial capabilities caching on first startup, by moving the cached capabilities to package.json so that on the first start up, the capabilities are available for different UI elements. 

Thanks for explaining this. The approach LGTM.
